processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -445,7 +445,9 @@ public int compare(Pair<Integer, PeekingIterator<String>> lhs, Pair<Integer, Pee
         }
         Indexed<String> indexed = dimValueLookups[i];
         if (useDirect) {
-          conversions[i] = ByteBuffer.allocateDirect(indexed.size() * Ints.BYTES).asIntBuffer();
+          int allocationSize = indexed.size() * Ints.BYTES;
+          log.info("Allocating dictionary merging direct buffer with size[%d]", allocationSize);
```

Comment:
Hm, there isn't a corresponding deallocation, I'll do that in a follow on PR to this